### PR TITLE
Corrected link targets in collections.rst

### PR DESCRIPTION
**Pull request by MSeifert04**
Fixes several link targets in [`library.collections.rst`](https://docs.python.org/library/collections.html).

There are 3 remaining "invalid" links:

- `list.append`, because there is no "target" for that method. There is something documented in [`tutorial/datastructures`](https://docs.python.org/3/tutorial/datastructures.html#more-on-lists) but these are not valid link targets.
- `defaultdict.get`, because it's not explicitly documented. Should it link to `dict.get`?
- `OrderedDict.update`, because it's not explicitly documented. Should it link to `dict.update`?

### CODE REVIEW

File: Doc/library/collections.rst [hunk 1]
```diff
@@ -357,21 +357,22 @@ or subtracting from an empty counter.
       restrictions on its keys and values.  The values are intended to be numbers
       representing counts, but you *could* store anything in the value field.
 
-    * The :meth:`most_common` method requires only that the values be orderable.
+    * The :meth:`~Counter.most_common` method requires only that the values be
```
Comment (serhiy-storchaka) [inline Doc/library/collections.rst]: Change also `:func:'most_common'` above to `:meth:'most_common'`.

File: Doc/library/collections.rst [hunk 2]
```diff
@@ -618,19 +619,19 @@ added elements by appending to the right and popping to the left::
             d.append(elem)
             yield s / n
 
-The :meth:`rotate` method provides a way to implement :class:`deque` slicing and
-deletion.  For example, a pure Python implementation of ``del d[n]`` relies on
-the :meth:`rotate` method to position elements to be popped::
+The :meth:`~deque.rotate` method provides a way to implement :class:`deque` slicing
+and deletion.  For example, a pure Python implementation of ``del d[n]`` relies on
+the :meth:`~deque.rotate` method to position elements to be popped::
```
Comment (serhiy-storchaka) [inline Doc/library/collections.rst]: Avoid using the same link multiple times in one paragraph. ` ``rotate()`` ` would look better.

File: Doc/library/collections.rst [hunk 3]
```diff
@@ -1035,8 +1036,8 @@ anywhere a regular dictionary is used.
 
 .. versionchanged:: 3.6
    With the acceptance of :pep:`468`, order is retained for keyword arguments
-   passed to the :class:`OrderedDict` constructor and its :meth:`update`
-   method.
+   passed to the :class:`OrderedDict` constructor and its
+   :meth:`update` method.
```
Comment (serhiy-storchaka) [inline Doc/library/collections.rst]: What is changed here?

File: Doc/library/collections.rst [hunk 4]
```diff
@@ -1175,6 +1176,11 @@ attribute.
     subclass) or an arbitrary sequence which can be converted into a string using
     the built-in :func:`str` function.
 
+    .. attribute:: data
```
Comment (serhiy-storchaka) [inline Doc/library/collections.rst]: Shouldn't be added the paragraph "In addition to supporting ..." as for UserDict and UserList?

File: Doc/library/collections.rst [hunk 5]
```diff
@@ -1175,6 +1176,11 @@ attribute.
     subclass) or an arbitrary sequence which can be converted into a string using
```
Comment (serhiy-storchaka) [inline Doc/library/collections.rst]: This phrase is outdated (remained from Python 2). Virtually any object (not just sequences) can be converted into a string using str(). However bytes is bad example, it emits a warning or an exception when converted to str.

**Review by serhiy-storchaka — COMMENTED** [review]
Mostly looks good, but I have some comments.

File: Doc/library/collections.rst (hunk 3 above)
Comment (MSeifert04) [reply Doc/library/collections.rst]: Nothing. 

I originally changed it to `dict.update` but then I wasn't sure if that was what was intended and reverted it again. Just forgot to revert all of the change.

**Review by serhiy-storchaka — APPROVED** [review]

File: Doc/library/collections.rst [hunk 6]
```diff
@@ -1165,15 +1166,22 @@ subclass directly from :class:`str`; however, this class can be easier
...
 
-    Class that simulates a string or a Unicode string object.  The instance's
+    Class that simulates a string object.  The instance's
     content is kept in a regular string object, which is accessible via the
     :attr:`data` attribute of :class:`UserString` instances.  The instance's
-    contents are initially set to a copy of *sequence*.  The *sequence* can
-    be an instance of :class:`bytes`, :class:`str`, :class:`UserString` (or a
-    subclass) or an arbitrary sequence which can be converted into a string using
-    the built-in :func:`str` function.
+    contents are initially set to a copy of *seq*.  The *seq* can
```
Comment (vadmium) [inline Doc/library/collections.rst]: “The `*seq*`” is missing a noun. I would either write “`*Seq*` can . . .” or “The `*seq*` argument can . . .”. Changing the parameter name makes this problem worse.

**Review by orsenthil — APPROVED** [review]
LGTM.

**Comment by serhiy-storchaka** [comment]
@rhettinger could you please make a look as the `collections` module maintainer?

**Comment by Mariatta** [comment]
I think this looks good, I just wonder if we need a ticket in b.p.o for this?

**Comment by rhettinger** [comment]
Please hold off on this for while.  I won't have a chance to go through every change in detail for a while.  Some of the changes like func->meth or meth->attr look fine; however, a number of the meth entries were intentionally not made into link target (i.e. I mention rotate() several times in a row, I don't want them to all be links -- this just garbages the appearance without making the docs better.  The double colon was left-off of the combination_with_replacement example on purpose because it includes sample output which isn't real code.  That said, I think it looks better with the double colon, so I think I'll put a # comment on the line or somesuch.    When proposing extensive edits, please don't rewrap the lines unless there is a major increase to line length, the rewrapping makes it harder to review.

**Comment by MSeifert04** [comment]
I haven't realized that this was an "extensive edit". If it would help I can seperate the different changes into seperate PRs (with accompanying issues on the tracker).

There are basically 4 changes here:

- wrong target `:meth:` instead of `:attr:` or `:func:` instead of `:meth:`
- missing `data` attribute and outdated documentation of `Userstring`
- the double colon when referencing `itertools.combinations`
- prepending the class to "not-linking" links

**Comment by MSeifert04** [comment]
How should I proceed with this PR?

**Comment by MSeifert04** [comment]
Anything I can or should do? 

It's a relatively small PR and it has been stalling for months now.

**Comment by MSeifert04** [comment]
@rhettinger Is it worth fixing the conflicts? I don't want to invest more time into this PR if you don't like it.

**Comment by MSeifert04** [comment]
@serhiy-storchaka Thanks for fixing the conflicts :) 

I still wonder if it's worth keeping it open. There seems to be no interest in merging it, or is just that @rhettinger has no time to review it?

**Comment by serhiy-storchaka** [comment]
All @rhettinger's comments looks addressed to me.

**Comment by MSeifert04** [comment]
There is one comment that I haven't addressed (I think) because I don't know which ones were meant:

>  however, a number of the meth entries were intentionally not made into link target (i.e. I mention rotate() several times in a row, I don't want them to all be links -- this just garbages the appearance without making the docs better.

**Comment by serhiy-storchaka** [comment]
It was addressed before it was written. There are no more than one links to the same target in one paragraph. If `rotate()` is mentioned several times in a row, only the first occurrence is a link.

**Comment by MSeifert04** [comment]
okay - so what is holding up the PR?

**Comment by MSeifert04** [comment]
@rhettinger Do you have time to review the PR?

**Pull request merged**